src/oop_ext/interface/_tests/test_interface.py

```diff
@@ -1077,3 +1077,39 @@ def AfterCaption(*args):
 
     assert Foo.GetCaption() == "Foo"
     assert Foo().GetValues("m") == [0.1, 10.0]
+
+
+def testGenericInterface() -> None:
+    """Generic interfaces need to support checking on both generic and non-generic implementers"""
+    from typing import FrozenSet, Generic, TypeVar
+
+    T = TypeVar("T", covariant=True)
+
+    class IFoo(Interface, Generic[T], TypeCheckingSupport):
+        # Class instance defined here as a workaround for this class to work in Python 3.6.
+        __abstractmethods__: FrozenSet = frozenset()
```

Comment:
Let's drop Python 3.6 first.

```suggestion
```

Reply:
To drop Python 3.6:

* Review CI: https://github.com/ESSS/oop-ext/blob/master/.github/workflows/main.yml
* Update `setup.py`:
  * https://github.com/ESSS/oop-ext/blob/9b6f94437a0eb1516eca82a7fdb47c6c11663efd/setup.py#L35
  * https://github.com/ESSS/oop-ext/blob/9b6f94437a0eb1516eca82a7fdb47c6c11663efd/setup.py#L47
* Update `tox.ini`: https://github.com/ESSS/oop-ext/blob/9b6f94437a0eb1516eca82a7fdb47c6c11663efd/tox.ini#L2
* Update `CHANGELOG`: https://github.com/ESSS/oop-ext/blob/master/CHANGELOG.rst
* Update `environment.devenv.yml`: https://github.com/ESSS/oop-ext/blob/9b6f94437a0eb1516eca82a7fdb47c6c11663efd/environment.devenv.yml#L7